File: src/types/note.ts
export type Note = {
  id: string;
  title: string;
  content: string;
  preview: string;
  folderId: string;
  folderName: string;
  tags: string[];
  isPinned: boolean;
  isFavorite: boolean;
  isDeleted: boolean;
  isArchived: boolean;
  isLocked: boolean;
  isUnlocked?: boolean;
  encryptedContent?: string | null;
  encryptedPreview?: string | null;
  encryptionNonce?: string | null;
  lockedAt?: string | null;
  createdAt: string;
  updatedAt: string;
};

export type Folder = {
  id: string;
  name: string;
  colorClass: string;
};

export type Attachment = {
  id: string;
  noteId: string;
  filename: string;
  originalPath?: string | null;
  storedPath: string;
  mimeType: string;
  fileSize: number;
  createdAt: string;
};

export type SidebarView =
  | "all"
  | "favorites"
  | "recent"
  | "archive"
  | "trash"
  | "graph"
  | "settings";

File: Android/app/src/main/java/com/lumonotes/app/domain/Note.kt
package com.lumonotes.app.domain

data class Note(
    val id: String,
    val title: String,
    val content: String,
    val preview: String,
    val folderId: String,
    val folderName: String,
    val tags: List<String>,
    val isPinned: Boolean,
    val isFavorite: Boolean,
    val isDeleted: Boolean,
    val isArchived: Boolean,
    val isLocked: Boolean,
    val encryptedContent: String?,
    val encryptedPreview: String?,
    val encryptionNonce: String?,
    val lockedAt: String?,
    val createdAt: String,
    val updatedAt: String,
)
